server/runtime.go

```diff
@@ -312,3 +343,285 @@ func (r *Runtime) invokeFunction(l *lua.LState, fn *lua.LFunction, ctx *lua.LTab
+		if err != nil {
+			return nil, err
+		}
+
+		var jsonEnvelope map[string]interface{}
+		if err = json.Unmarshal([]byte(strEnvelope), &jsonEnvelope); err != nil {
+			return nil, err
+		}
+
+		for k, v := range jsonEnvelope {
+			lt.RawSetString(k, convertValue(l, v))
+		}
+	}
+
+	return lt, nil
+}
+
+func ConvertLTableToEnvelope(l *lua.LState, jsonpbUnmarshaler *jsonpb.Unmarshaler, lt *lua.LTable, envelope *Envelope) (*Envelope, error) {
+	switch envelope.Payload.(type) {
+	case *Envelope_GroupsRemove:
+		groupsRemove := lt.RawGetString("GroupsRemove")
+		if groupsRemove.Type() != lua.LTTable {
+			return nil, errors.New("Invalid payload in GroupsRemove conversion from script runtime")
+		}
+		groupIds := groupsRemove.(*lua.LTable).RawGetString("GroupIds")
+		if groupIds.Type() != lua.LTTable {
+			return nil, errors.New("Invalid payload in GroupsRemove conversion from script runtime")
+		}
+		ids := make([][]byte, 0)
+		var err error
+		groupIds.(*lua.LTable).ForEach(func(k lua.LValue, v lua.LValue) {
+			if v.Type() != lua.LTString {
+				err = errors.New("Invalid Group ID in GroupsRemove conversion from script runtime")
+				return
+			}
+			gid, e := uuid.FromString(v.String())
+			if e != nil {
+				err = errors.New("Invalid Group ID in GroupsRemove conversion from script runtime")
+				return
+			}
+			ids = append(ids, gid.Bytes())
+		})
+		if err != nil {
+			return nil, err
+		}
+		envelope.GetGroupsRemove().GroupIds = ids
+	case *Envelope_GroupsJoin:
+		groupsJoin := lt.RawGetString("GroupsJoin")
+		if groupsJoin.Type() != lua.LTTable {
+			return nil, errors.New("Invalid payload in GroupsJoin conversion from script runtime")
+		}
+		groupIds := groupsJoin.(*lua.LTable).RawGetString("GroupIds")
+		if groupIds.Type() != lua.LTTable {
+			return nil, errors.New("Invalid payload in GroupsJoin conversion from script runtime")
+		}
+		ids := make([][]byte, 0)
+		var err error
+		groupIds.(*lua.LTable).ForEach(func(k lua.LValue, v lua.LValue) {
+			if v.Type() != lua.LTString {
+				err = errors.New("Invalid Group ID in GroupsJoin conversion from script runtime")
+				return
+			}
+			gid, e := uuid.FromString(v.String())
+			if e != nil {
+				err = errors.New("Invalid Group ID in GroupsJoin conversion from script runtime")
+				return
+			}
+			ids = append(ids, gid.Bytes())
+		})
+		if err != nil {
+			return nil, err
+		}
+		envelope.GetGroupsJoin().GroupIds = ids
+	case *Envelope_GroupsLeave:
+		groupsLeave := lt.RawGetString("GroupsLeave")
+		if groupsLeave.Type() != lua.LTTable {
+			return nil, errors.New("Invalid payload in GroupsLeave conversion from script runtime")
+		}
+		groupIds := groupsLeave.(*lua.LTable).RawGetString("GroupIds")
+		if groupIds.Type() != lua.LTTable {
+			return nil, errors.New("Invalid payload in GroupsLeave conversion from script runtime")
+		}
+		ids := make([][]byte, 0)
+		var err error
+		groupIds.(*lua.LTable).ForEach(func(k lua.LValue, v lua.LValue) {
+			if v.Type() != lua.LTString {
```

Comment:
I think this check can be removed since `uuid.FromString` will produce an error if `v` is not a valid UUID string.